wasm/jit/jit_amd64_test.go

```diff
@@ -3877,6 +3887,192 @@ func TestAmd64Compiler_compileRem(t *testing.T) {
 	})
 }
 
+func TestAmd64Compiler_compileF32DemoteFromF64(t *testing.T) {
+	for _, v := range []float64{
+		0, 100, -100, 1, -1,
+		100.01234124, -100.01234124, 200.12315,
+		math.MaxFloat32,
+		math.SmallestNonzeroFloat32,
+		math.MaxFloat64,
+		math.SmallestNonzeroFloat64,
+		6.8719476736e+10,  /* = 1 << 36 */
+		1.37438953472e+11, /* = 1 << 37 */
+		math.Inf(1), math.Inf(-1), math.NaN(),
+	} {
+		t.Run(fmt.Sprintf("%f", v), func(t *testing.T) {
+			compiler := requireNewCompiler(t)
+			compiler.initializeReservedRegisters()
+
+			// Setup the demote target.
+			err := compiler.compileConstF64(&wazeroir.OperationConstF64{Value: v})
+			require.NoError(t, err)
+
+			err = compiler.compileF32DemoteFromF64()
+			require.NoError(t, err)
+
+			// To verify the behavior, we release the value
+			// to the stack.
+			compiler.releaseAllRegistersToStack()
+			compiler.returnFunction()
+
+			// Generate and run the code under test.
+			code, _, err := compiler.generate()
+			require.NoError(t, err)
+			eng := newEngine()
+			jitcall(uintptr(unsafe.Pointer(&code[0])), uintptr(unsafe.Pointer(eng)), 0)
+
+			// Check the result.
+			require.Equal(t, uint64(1), eng.stackPointer)
+			if math.IsNaN(v) {
+				require.True(t, math.IsNaN(float64(math.Float32frombits(uint32(eng.stack[eng.stackPointer-1])))))
```

Comment:
haha quite a mouthful.  if any part of this conversion is common, maybe extract a function.. perhaps the `math.Float32frombits(uint32` as `float32frombits`

Reply:
done https://github.com/tetratelabs/wazero/pull/121/commits/f495b1d81c633831c6078cbd32a03a3122cbc3cc